packages/commons/src/fhir/transformers/index.ts

```diff
@@ -647,6 +647,11 @@ function createDateOfMarriageBuilder(resource: Patient, fieldValue: string) {
   if (!resource.extension) {
     resource.extension = []
   }
+
+  resource.extension = resource.extension.filter(
```

Comment:
Shouldn't these be also made with setExtension? You might have to modify the function a bit but a signature like

```ts
resource.extension = setExtension(
  resource.extension, 
  `${OPENCRVS_SPECIFICATION_URL}extension/date-of-marriage`, 
  { valueDateTime: fieldValue }
)
```

could just make sense, right?

Reply:
Yeah, it would. We'll have to restructure the `setExtension` function then.